frontend/src/components/nodes/Nodes.tsx

```diff
@@ -28,7 +45,55 @@ export default class extends React.Component<OuterProps> {
   autoRefreshNodes = autoRefreshHandler(Nodes, this.config);
 
   render() {
-    return <this.autoRefreshNodes />;
+    return (
+      <>
+        <div>
+          <Row>
+            <Button
+              onClick={this.fetchValidatorNodes}
```

Comment:
You are simulating pages here. Instead of abusing state, let's just have this encoded as the page props, so:

* `/nodes/validators`
* `/nodes/non-validators`

Rename `src/pages/nodes/index.jsx` into `src/pages/nodes/[role].jsx` and handle `role` in `getInitialProps` (you can see how we handle `hash` in `src/pages/blocks/[hash].jsx`), and pass it down to the `Nodes.tsx`.